### PR TITLE
feat/95 [Admin Side - API] Add Excercise Library Mangament

### PR DESCRIPTION
**Pull request by QuocHuy178**
Testing result:
image

Before Add Exercise:
image

After Add Exercise:
image

### CODE REVIEW

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            try
+            {
+                var checkId = await CheckClassExist(request.ClassId);
+
+                if (!checkId)
+                    return new BaseResponseModel(HttpStatusCode.NotFound, "Class Id invalid");
+
+
+                var query = @"INSERT INTO Exercise (ClassId, Name, Image, IsActive, CreatedDate, CreatedBy, UpdatedBy, UpdatedDate)
+                                VALUES (@ClassId, @ExName, @ExImage, @IsActive, @CreatedDate, @CreatedBy, @UpdateBy, @UpdatedDate)";
```
Comment (ducvp0702) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs]: Should check the Name of a challenge if it exits so we can't create

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            _connection = connection;
+        }
+        #endregion
+
+        public async Task<BaseResponseModel> CreateExerciseLib(CreateExerciseLibRequestModel request)
+        {
+            try
+            {
+                var checkId = await CheckClassExist(request.ClassId);
```
Comment (ducvp0702) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs:35]: CheckClassExist should in repository

File: src/Services/GTT/GTT.Api/ExerciseLibraryManagement/CreateExerciseLibrary.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = Routes.ExerciseLibrary)] HttpRequestData req)
+        {
+            try
+            {
+                _logger.LogInformation("C# HTTP trigger function CreateExerciseLibrary request.");
+                var requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+                var data = JsonConvert.DeserializeObject<CreateExerciseLibRequestModel>(requestBody);
+                var result = await _mediator.Send(new CreateExerciseLib.Command(data));
+                var respone = req.CreateResponse();
+                await respone.WriteAsJsonAsync(result, result.Status == HttpStatusCode.OK ? HttpStatusCode.OK : HttpStatusCode.BadRequest);
```
Comment (phantuananh18) [inline src/Services/GTT/GTT.Api/ExerciseLibraryManagement/CreateExerciseLibrary.cs]: Remove the ternary operator, you should return result.Status

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            }
+        }
+
+        private async Task<bool> CheckClassExist(int Id)
+        {
+            var checkClassId = @"SELECT * FROM CLASS
+                                    WHERE ClassId = @Id";
+            var resultCheck = await _connection.QueryFirstOrDefaultAsync(checkClassId, new { Id });
+
+            return resultCheck != null ? true : false;
```
Comment (phantuananh18) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs]: using try catch

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        public async Task<BaseResponseModel> CreateExerciseLib(CreateExerciseLibRequestModel request)
+        {
+            try
+            {
+                var checkId = await CheckClassExist(request.ClassId);
+
+                if (!checkId)
+                    return new BaseResponseModel(HttpStatusCode.NotFound, "Class Id invalid");
+
+                var queryCheckName = @"SELECT * FROM Exercise e
```
Comment (phantuananh18) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs]: Don't use * in query, you should write column clearly

File: src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                                            WHERE e.Name = @ExName";
+
+                var query = @"INSERT INTO Exercise (ClassId, Name, Image, IsActive, CreatedDate, CreatedBy, UpdatedBy, UpdatedDate)
+                                VALUES (@ClassId, @ExName, @ExImage, @IsActive, @CreatedDate, @CreatedBy, @UpdateBy, @UpdatedDate)
+                                DECLARE @exerciseId int                             
+                                SET @exerciseId = SCOPE_IDENTITY()
+                                SELECT * FROM Exercise WHERE ExerciseId = @exerciseId";
```
Comment (phantuananh18) [inline src/Services/GTT/shared/GTT.Infrastructure/Repositories/ExerciseLibRepository.cs]: Don't use * in query, you should write column clearly

**Pull request closed without merging**